src/test/app/AccountDelete_test.cpp

```diff
@@ -912,6 +912,275 @@ class AccountDelete_test : public beast::unit_test::suite
+            env.close();
+
+            // becky attempts to delete her account, but alice won't take her
+            // XRP, so the delete is blocked.
+            env(acctdelete(becky, alice),
+                fee(acctDelFee),
+                ter(tecNO_PERMISSION));
+
+            // becky use empty credentials and can't delete account
+            env(acctdelete(becky, alice),
+                fee(acctDelFee),
+                credentials::IDs({}),
+                ter(temMALFORMED));
+
+            // becky use bad credentials and can't delete account
+            env(acctdelete(becky, alice),
+                credentials::IDs(
+                    {"48004829F915654A81B11C4AB8218D96FED67F209B58328A72314FB6E"
+                     "A288BE4"}),
+                fee(acctDelFee),
+                ter(tecBAD_CREDENTIALS));
+            env.close();
+
+            // becky use credentials and can delete account
+            env(acctdelete(becky, alice),
+                credentials::IDs({credIdx}),
+                fee(acctDelFee));
+            env.close();
+
+            // check that credential object deleted too
+            auto const jNoCred =
+                credentials::ledgerEntryCredential(env, becky, carol, credType);
+            BEAST_EXPECT(
+                jNoCred.isObject() && jNoCred.isMember(jss::result) &&
+                jNoCred[jss::result].isMember(jss::error));
+
+            {
+                Account const john{"john"};
+
+                env.fund(XRP(100000), john);
+                env.close();
+
+                auto jv = credentials::create(john, carol, credType);
+                uint32_t const t = env.now().time_since_epoch().count() + 20;
+                jv[sfExpiration.jsonName] = t;
+                env(jv);
+                env.close();
+                env(credentials::accept(john, carol, credType));
+                env.close();
+                jv = credentials::ledgerEntryCredential(
+                    env, john, carol, credType);
+                std::string const credIdx =
+                    jv[jss::result][jss::index].asString();
+
+                incLgrSeqForAccDel(env, john);
+
+                // credentials are expired
+                // john use credentials but can't delete account
+                env(acctdelete(john, alice),
+                    credentials::IDs({credIdx}),
+                    fee(acctDelFee),
+                    ter(tecEXPIRED));
+                env.close();
+
+                {
+                    // check that credential object deleted
+                    auto jv = credentials::ledgerEntryCredential(
+                        env, john, carol, credType);
+                    BEAST_EXPECT(
+                        jv.isObject() && jv.isMember(jss::result) &&
+                        jv[jss::result].isMember(jss::error));
+                }
+            }
+        }
+
+        {
+            testcase("Deleting Issuer delete credentials");
```

Comment:
IMO this and the next test (checking whether credentials are deleted properly when a subject/issuer is deleted) should be in a separate function.

Reply:
moved to `testDeleteCredentialsOwner`